File: libensemble/tests/controller_tests/simdir/my_simjob.c
#include <unistd.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <mpi.h>

int main(int argc, char **argv) 
{
    int ierr, num_procs, rank, delay;
        
    delay=3;
    if (argc >=3) {
        if (strcmp( argv[1],"sleep") ==0 ) {
            delay = atoi(argv[2]);
        }
    }
    ierr = MPI_Init(&argc, &argv);
    ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    ierr = MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
    
    printf("Hello world sleeping for %d seconds on rank %d of %d\n",delay, rank,num_procs);  
    sleep(delay);
    //printf("Woken up on rank %d of %d\n",rank,num_procs);  
    ierr = MPI_Finalize();   
    return(0);
}

File: libensemble/tests/controller_tests/simdir/my_simjob.f90
module Fortran_Sleep
use, intrinsic :: iso_c_binding, only: c_int
implicit none
interface
   !should be unsigned int ... not available in Fortran
   !OK until highest bit gets set.
   function FortSleep(seconds)  bind(C, name="sleep")
       import
       integer (c_int) :: FortSleep
       integer (c_int), intent (in), VALUE :: seconds
   end function FortSleep
end interface

end module Fortran_Sleep

program hello_fortran
  use mpi
  use, intrinsic :: iso_c_binding, only: c_int
  use Fortran_Sleep

  implicit none

  integer (kind = 4) :: err
  integer (kind = 4) :: rank
  integer (kind = 4) :: nprocs
  character(len=32)  :: arg
  character(len=32)  :: arglist(2)
  integer (c_int)    :: delay,dummy
  integer            :: i

  delay=5
  dummy = FortSleep(delay)


  !init in case get no args
  DO i = 1, 2
    arglist(i) = "None"
  END DO

  DO i = 1, iargc()
    CALL getarg(i, arg)
    arglist(i) = arg
  END DO

  call mpi_init(err)
  call mpi_comm_size (MPI_COMM_WORLD, nprocs, err)
  call mpi_comm_rank (MPI_COMM_WORLD, rank, err)



  write(*,100) rank, nprocs, trim(arglist(1)), trim(arglist(2))

  call MPI_Finalize (err)

  100 FORMAT('Hello World from proc: ',i3,' of',i3,'  arg1: ',a,'  arg2: ',a)
end program hello_fortran
